core/permission/user.go

```diff
@@ -0,0 +1,26 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package permission
+
+import "fmt"
+
+// UserPermissionAccess represents the access a user has to a permission.
+type UserPermissionAccess struct {
```

Comment:
I'm not sold on this name, but `UserAccess` has already been taken. We can probably change later if required.

Reply:
You're competing with domain/permission.UserAccessSpec which does the same thing. https://github.com/juju/juju/blob/318bf1f1755894295b95983ed0e1eb40bdc2f3d3/domain/permission/types.go#L49

Since it'll be shared now - it does belong in core, but we only need one, not two.